spras/analysis/summary.py

```diff
@@ -50,8 +51,44 @@ def summarize_networks(file_paths: Iterable[Path], node_table: pd.DataFrame, alg
         number_edges = nw.number_of_edges()
         ncc = nx.number_connected_components(nw)
 
+        # Save the max/median degree, average clustering coeffienct, and density
+        if number_nodes <= 1 or number_edges == 0:
+            max_degree = 0
+            median_degree = 0.0
+            density = 0.0
+        else:
+            degrees = [deg for _, deg in nw.degree()]
+            max_degree = max(degrees)
+            median_degree = median(degrees)
+            density = nx.density(nw)
+
+        # Save the max diameter
+        cc = list(nx.connected_components(nw))
+        diameters = [
+            nx.diameter(nw.subgraph(c).copy()) if len(c) > 1 else 0
+            for c in cc
+        ]
+        max_diameter = max(diameters, default=0)
+
+        # # Save the max average clustering coefficient
+        # component_accs = [
+        #     nx.average_clustering(nw.subgraph(c).copy()) for c in cc if len(c) > 2
+        # ]
+        # acc_max = max(component_accs, default=0.0)
+
+        # Save the average path lengths
+        avg_path_lengths = [
+            nx.average_shortest_path_length(nw.subgraph(c).copy()) if len(c) > 1 else 0.0
+            for c in nx.connected_components(nw)
+        ]
+
+        if avg_path_lengths:
```

Comment:
They do the same thing (I hope), but this suggestion has clearer intent.

```suggestion
        if len(avg_path_lengths) != 0:
```